_search-plugins/inner-hits.md

```diff
@@ -0,0 +1,750 @@
+              "rating" : 5
+            },
+            {
+              "user" : "Eve",
+              "comment" : "Good value",
+              "rating" : 4
+            }
+          ]
+        },
+        "inner_hits" : {
+          "top_reviews" : {
+            "hits" : {
+              "total" : {
+                "value" : 2,
+                "relation" : "eq"
+              },
+              "max_score" : null,
+              "hits" : [
+                {
+                  "_index" : "products",
+                  "_id" : "2",
+                  "_nested" : {
+                    "field" : "reviews",
+                    "offset" : 0
+                  },
+                  "_score" : null,
+                  "_source" : {
+                    "rating" : 5,
+                    "comment" : "Very good",
+                    "user" : "Dave"
+                  },
+                  "sort" : [
+                    5
+                  ]
+                },
+                {
+                  "_index" : "products",
+                  "_id" : "2",
+                  "_nested" : {
+                    "field" : "reviews",
+                    "offset" : 1
+                  },
+                  "_score" : null,
+                  "_source" : {
+                    "rating" : 4,
+                    "comment" : "Good value",
+                    "user" : "Eve"
+                  },
+                  "sort" : [
+                    4
+                  ]
+                }
+              ]
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+### Example of `inner_hits` options with Parent-Join Relationship
+
+1. Create index and mappings:
+    ```
+    PUT /company
+    {
+      "mappings": {
+        "properties": {
+          "my_join_field": {
+            "type": "join",
+            "relations": {
+              "employee": "task"
+            }
+          },
+          "name": { "type": "text" },
+          "description": {
+            "type": "text",
+            "fields": {
+              "keyword": { "type": "keyword" }
+            }
+          }
+        }
+      }
+    }
+    ```
+
+2. Index data:
+    ```
+    # Index a parent document
+    PUT /company/_doc/1
+    {
+      "name": "Alice",
+      "my_join_field": "employee"
+    }
+    
+    # Index child documents
+    PUT /company/_doc/2?routing=1
+    {
+      "description": "Complete the project",
+      "my_join_field": {
+        "name": "task",
+        "parent": "1"
+      }
+    }
+    
+    PUT /company/_doc/3?routing=1
+    {
+      "description": "Prepare the report",
+      "my_join_field": {
+        "name": "task",
+        "parent": "1"
+      }
+    }
+    ```
+
+3. Query with Inner Hits Options:
+    ```
+    GET /company/_search
+    {
+      "query": {
+        "has_child": {
+          "type": "task",
+          "query": {
+            "match": { "description": "project" }
+          },
+          "inner_hits": {
```

Comment:
it would be nice if we can return 2 documents to show in the result that they are sorted by `description` `asc`.